test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfiguration.cs

```diff
@@ -48,7 +54,60 @@ public static EdgeConfiguration Create(string deviceId, IEnumerable<ModuleConfig
             // Make a copy
             config = JsonConvert.DeserializeObject<ConfigurationContent>(JsonConvert.SerializeObject(config));
 
-            return new EdgeConfiguration(deviceId, names, images, config);
+            // Build the object we'll use later to verify the deployment
+            var edgeAgent = modules.Where(m => m.Name == ModuleName.EdgeAgent).FirstOrDefault()
+                ?? new ModuleConfiguration();
+            var desired = JObject
+                .FromObject(edgeAgent.DesiredProperties);
+
+            var reported = new Dictionary<string, object>
+            {
+                ["systemModules"] = desired
+                            .Value<JObject>("systemModules")
+                            .Children<JProperty>()
+                            .ToDictionary(
+                                p => p.Name,
+                                p => CreateExpectedModuleConfig((JObject)p.Value))
+            };
+
+            if (desired.ContainsKey("modules"))
+            {
+                reported["modules"] = desired
+                    .Value<JObject>("modules")
+                    .Children<JProperty>()
+                    .ToDictionary(
+                        p => p.Name,
+                        p => CreateExpectedModuleConfig((JObject)p.Value));
+            }
+
+            var expected = new
+            {
+                properties = new
+                {
+                    reported = reported
+                }
+            };
+
+            return new EdgeConfiguration(deviceId, names, images, config, expected);
+        }
+
+        static object CreateExpectedModuleConfig(JObject source)
+        {
+            var image = source.SelectToken($"settings.image"); // not optional
```

Comment:
If this doesn't exist in the source, will there be an exception?

Reply:
I believe so, either at this line, or later on when we try to dereference null. But that's good; the exception will bubble up and the test will fail.